src/sema/namespace.rs

```diff
@@ -309,36 +309,25 @@ impl Namespace {
     pub fn resolve_contract_with_namespace(
         &mut self,
         file_no: usize,
-        expr: &pt::Expression,
+        name: &pt::IdentifierPath,
         diagnostics: &mut Vec<Diagnostic>,
     ) -> Result<usize, ()> {
-        let (namespace, id, dimensions) = self.expr_to_type(file_no, None, expr, diagnostics)?;
-
-        if !dimensions.is_empty() {
+        let (id, namespace) = if let Some((id, namespace)) = name.identifiers.split_last() {
+            (id, namespace.iter().collect())
```

Comment:
`.iter().collect()` can be `.to_vec()`, I think (not terribly important)

Reply:
I thought about this too, but I don't if this is possible here as `split_last` returns `(&T, &[T])` and I don't know if its possible to easily convert a `&[T]` to a `[&T]` to convert directly to a `Vec<&T>` like the call signature expects

Reply:
Right, I had not thought of that. Looks great!